diff --git a/mediagoblin/media_types/tools.py b/mediagoblin/media_types/tools.py
index 3bc56e2..95b7d23 100644
--- a/mediagoblin/media_types/tools.py
+++ b/mediagoblin/media_types/tools.py
@@ -35,8 +35,10 @@ def discover(src):
     # import, or an exception will be raised.
     import gi
     gi.require_version('Gst', '1.0')
-    from gi.repository import GObject, Gst, GstPbutils, GLib
+    from gi.repository import GObject, Gst, GLib
     Gst.init(None)
+    # init before import to work around https://bugzilla.gnome.org/show_bug.cgi?id=736260
+    from gi.repository import GstPbutils 
 
     _log.info('Discovering {0}...'.format(src))
     uri = 'file://{0}'.format(src)
diff --git a/mediagoblin/media_types/video/transcoders.py b/mediagoblin/media_types/video/transcoders.py
index 3c060fd..8a27eac 100644
--- a/mediagoblin/media_types/video/transcoders.py
+++ b/mediagoblin/media_types/video/transcoders.py
@@ -31,8 +31,10 @@ sys.argv = []
 
 import gi
 gi.require_version('Gst', '1.0')
-from gi.repository import GObject, Gst, GstPbutils
+from gi.repository import GObject, Gst
 Gst.init(None)
+# init before import to work around https://bugzilla.gnome.org/show_bug.cgi?id=736260
+from gi.repository import GstPbutils 
 
 sys.argv = old_argv
 import struct
